src/sentry/static/sentry/app/components/charts/components/legend.tsx

```diff
@@ -4,20 +4,26 @@ import 'echarts/lib/component/legendScroll';
 import {EChartOption} from 'echarts';
 
 import BaseChart from 'app/components/charts/baseChart';
+import {Theme} from 'app/utils/theme';
 
 import {truncationFormatter} from '../utils';
 
 type ChartProps = React.ComponentProps<typeof BaseChart>;
 
-export default function Legend(props: ChartProps['legend'] = {}): EChartOption.Legend {
-  const {truncate, ...rest} = props ?? {};
+export default function Legend(
+  props: ChartProps['legend'] & {theme?: Theme} = {}
```

Comment:
Would it be better to add `{theme?: Theme}` in `BaseChart`?

https://github.com/getsentry/sentry/blob/7f6720837cf1f69f366d0be319cd5710f6c362af/src/sentry/static/sentry/app/components/charts/baseChart.tsx#L126

Reply:
imo no because theme is passed from BaseChart to `Legend` here (and not from a chart implementing BaseChart).